src/sr/discord_bot/commands/pin.py

```diff
@@ -0,0 +1,51 @@
+from typing import cast, TYPE_CHECKING
+
+import discord
+from discord import app_commands
+
+if TYPE_CHECKING:
+    from sr.discord_bot.bot import BotClient
+
+@app_commands.command(
+    name='pin',
+    description='Pin a message to the channel'
+)
+@app_commands.describe(
+    message_url='Link of the message to pin',
+)
+@app_commands.rename(message_url='message')
+async def pin_message(
+    interaction: discord.interactions.Interaction["BotClient"],
+    message_url: str,
+) -> None:
+    """Pin a message to the channel."""
+    channel = cast(discord.TextChannel, interaction.channel)
+    message_id = int(message_url.split('/')[-1])
```

Comment:
What sort of error handling do we want here if the input doesn't match the expected format? (i.e: do we want to handle the resulting `ValueError`?)
Related: should we e.g: ensure that the given url is actually to Discord?